### PR TITLE
[v0.26] Make SubscriptionSpec.Delivery mutable

### PR DESCRIPTION
**Pull request by pierDipi**
Cherry-pick of https://github.com/knative/eventing/pull/6139/files

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

/cc @maschmid @matzew

### CODE REVIEW

File: pkg/apis/messaging/v1/subscription_validation_test.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 	"github.com/google/go-cmp/cmp"
 	"knative.dev/pkg/apis"
 	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/ptr"
```
Comment (matzew) [inline pkg/apis/messaging/v1/subscription_validation_test.go:25]: why are we here using different `Int32()` than the other versions?

Comment (pierDipi) [reply pkg/apis/messaging/v1/subscription_validation_test.go:25]: pointer.Int32 was added after, so it's not available.

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**